### ISSUE TITLE
[Bug] Terrastorage sort breaks playerhead data

### ISSUE THREAD

**Issue by DikerdodlePlays** [TimVincii/Terrastorage#7]
### What happened?

Using Terrastorage 1.1.1 on Minecraft 1.21.1 along with a number of quality of life / performance mods and datapacks (no content adding mods). Specifically using some [Vanilla Tweaks](https://vanillatweaks.net/) datapacks that make playerheads with custom block textures and mob head textures. When I went to use the item sorting feature on my chest of playerheads, it deleted much of the data of the playerheads in order to stack them together. 

Before clicking sort button: ![2024-10-27_21 29 41](https://github.com/user-attachments/assets/dd25f511-5756-4855-b435-50659bff2c94)
After: ![2024-10-27_21 29 47](https://github.com/user-attachments/assets/fc60520c-c05a-41dc-bcbc-6a0fe40b7dc8)

### Terrastorage Version

1.1.1

### Fabric API Version

0.107.0+1.21.1

### Minecraft Version

1.21.1

### Other Mods

```shell
Auto Clicker (https://modrinth.com/mod/r8axuw4u) [21.1.0]
Blur+ (Fabric) (https://modrinth.com/mod/NK39zBp2) [5.0.0]
Cherished Worlds (https://modrinth.com/mod/3azQ6p0W) [10.1.0+1.21.1]
Concurrent Chunk Management Engine (https://modrinth.com/mod/VSNURh3q) [0.3.0+alpha.0.293+1.21.1]
Enhanced Block Entities (https://modrinth.com/mod/OVuFYfre) [0.10.2+1.21]
EntityCulling (https://modrinth.com/mod/NNAgCjsB) [1.7.0]
Fabric API (https://modrinth.com/mod/P7dR8mSH) [0.107.0+1.21.1]
Fabric Language Kotlin (https://modrinth.com/mod/Ha28R6CL) [1.12.3+kotlin.2.0.21]
FerriteCore (https://modrinth.com/mod/uXXizFIs) [7.0.2-hotfix]
ImmediatelyFast (https://modrinth.com/mod/5ZwdcRci) [1.3.0+1.21.1]
Indium (https://modrinth.com/mod/Orvt0mRa) [1.0.35+mc1.21]
Iris (https://modrinth.com/mod/YL57xq9U) [1.8.0-beta.4+mc1.21.1]
Lithium (https://modrinth.com/mod/gvQqBUqZ) [0.14.0-snapshot+mc1.21.1-build.88]
MaLiLib (https://modrinth.com/mod/GcWjdA9I) [0.21.0]
MiniHUD (https://modrinth.com/mod/UMxybHE8) [0.32.50]
Mod Menu (https://modrinth.com/mod/mOgUt4GM) [11.0.3]
Mouse Tweaks (https://modrinth.com/mod/aC3cM3Vq) [2.26]
Placeholder API (https://modrinth.com/mod/eXts2L7r) [2.4.1+1.21]
Replay Mod (https://modrinth.com/mod/Nv2fQJo5) [1.21-2.6.19]
Sodium (https://modrinth.com/mod/AANobbMI) [0.6.0-beta.2+mc1.21.1]
Sodium Extra (https://modrinth.com/mod/PtjYWJkn) [0.6.0-beta.3+mc1.21.1]
Terrastorage (https://modrinth.com/mod/QKXofFZC) [1.1.1]
Xaero's Minimap (https://modrinth.com/mod/1bokaNcj) [24.5.0]
Xaero's World Map (https://modrinth.com/mod/NcUtCpym) [1.39.0]
YetAnotherConfigLib (https://modrinth.com/mod/1eAoo2KR) [3.6.1+1.21-fabric]
Zoomify (https://modrinth.com/mod/w7ThoJFB) [2.14.2+1.21.1]
```

**Comment by TimVincii** [comment]
Hey, thank you for taking the time to report this bug, and for providing this level of detail about it, appreciate it very much!

The issue was that item sorting completely ignored any component data that was unique to an item stack, and instead focused on the item of the stack only. While all other Terrastorage features take component data into account, I unfortunately overlooked this in the item sorting logic, sorry about that.

I just released version 1.1.2, which addresses this issue:
* [[Modrinth] Terrastorage 1.1.2+1.20-1.20.1](https://modrinth.com/mod/terrastorage/version/1.1.2+1.20-1.20.1)
* [[CurseForge] Terrastorage 1.1.2+1.20-1.20.1](https://www.curseforge.com/minecraft/mc-mods/terrastorage/files/5854403)

I also went ahead and tested it with the same Vanilla Tweaks player heads and mob heads datapacks that you were using, and everything seems to be working properly now.

With that said, I will now close this issue as completed, if you run into any other problems, don’t hesitate to submit a new issue.

Once again, thank you for your help in reporting this bug, and I wish you a great day!
